testing/app.go

```diff
@@ -1,9 +1,135 @@
 package ibctesting
 
 import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	capabilitykeeper "github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+
 	"github.com/cosmos/ibc-go/modules/core/keeper"
+	"github.com/cosmos/ibc-go/testing/simapp"
 )
 
+var DefaultTestingAppInit func() (TestingApp, map[string]json.RawMessage) = SetupTestingApp
+
 type TestingApp interface {
-	IBCKeeper() keeper.Keeper
+	abci.Application
+
+	// ibc-go additions
+	GetBaseApp() *baseapp.BaseApp
+	GetStakingKeeper() stakingkeeper.Keeper
+	GetIBCKeeper() *keeper.Keeper
+	GetScopedIBCKeeper() capabilitykeeper.ScopedKeeper
```

Comment:
In theory, app devs just need to add these 4 functions to their app.go and implement `DefaultTestingAppInit` func